src/framework/components/model/component.js

```diff
@@ -24,65 +26,121 @@ import { Component } from '../component.js';
  */
 class ModelComponent extends Component {
     /**
-     * Create a new ModelComponent instance.
-     *
-     * @param {ModelComponentSystem} system - The ComponentSystem that created this Component.
-     * @param {Entity} entity - The Entity that this Component is attached to.
+     * @type {string}
+     * @private
      */
-    constructor(system, entity) {
-        super(system, entity);
+    _type = 'asset';
 
-        this._type = 'asset';
-
-        /**
-         * @type {number}
-         * @private
-         */
-        this._asset = null;
-        /**
-         * @type {Model}
-         * @private
-         */
-        this._model = null;
-
-        this._mapping = {};
-
-        this._castShadows = true;
-        this._receiveShadows = true;
-
-        /**
-         * @type {Asset}
-         * @private
-         */
-        this._materialAsset = null;
-        this._material = system.defaultMaterial;
+    /**
+     * @type {Asset|number|null}
+     * @private
+     */
+    _asset = null;
 
-        this._castShadowsLightmap = true;
-        this._lightmapped = false;
-        this._lightmapSizeMultiplier = 1;
-        this._isStatic = false;
+    /**
+     * @type {Model|null}
+     * @private
+     */
+    _model = null;
 
-        this._layers = [LAYERID_WORLD]; // assign to the default world layer
-        this._batchGroupId = -1;
+    /* eslint-disable jsdoc/check-types */
+    /**
+     * @type {Object.<string, number>}
+     * @private
+     */
+    _mapping = {};
+    /* eslint-enable jsdoc/check-types */
 
-        /**
-         * @type {BoundingBox}
-         * @private
-         */
-        this._customAabb = null;
+    /**
+     * @type {boolean}
+     * @private
+     */
+    _castShadows = true;
 
-        this._area = null;
+    /**
+     * @type {boolean}
+     * @private
+     */
+    _receiveShadows = true;
 
-        this._assetOld = 0;
-        this._materialEvents = null;
-        this._dirtyModelAsset = false;
-        this._dirtyMaterialAsset = false;
+    /**
+     * @type {Asset|number|null}
+     * @private
+     */
+    _materialAsset = null;
+
+    /**
+     * @type {Material}
+     * @private
+     */
+    _material;
+
+    /**
+     * @type {boolean}
+     * @private
+     */
+    _castShadowsLightmap = true;
 
-        this._clonedModel = false;
+    /**
+     * @type {boolean}
+     * @private
+     */
+    _lightmapped = false;
 
-        // #if _DEBUG
-        this._batchGroup = null;
-        // #endif
+    /**
+     * @type {number}
+     * @private
+     */
```

Comment:
I just wonder if these could be done on small number of lines? Perhaps
```
/** @type { number } */
/** @private */
```

Reply:
Unfortunately not. That's actually the minimum number of lines that will work for 2 tags. But the good news is that when we adopt private members/functions (`#`), it will collapse to a single line. I'm hoping we do that relatively soon.